File: analyse/python/opgave_16.py
# Opgave 16
# Volgens A-model

import numpy as np
import lib.print_m as pm
import lib.models as models

# Gegeven
y = np.matrix('78.83; 120.81; 38.71; 36.47; 80.61; 36.51')
A = np.matrix( '\
    0 1 0 0; \
    0 0 1 0; \
    -1 1 0 0; \
    0 0 -1 1; \
    -1 0 1 0; \
    0 0 -1 1 \
    ')

# Vereffening volgens A-model
xdakje, ydakje, edakje = models.amodel(A, y)

uitvoer = 'Geschatte waarden voor parameters xdakje:'
print(uitvoer)
pm.print_m('xdakje', xdakje, 2)
print('')

uitvoer = 'Geschatte waarden voor toevallige afwijkingen edakje:'
print(uitvoer)
pm.print_m('edakje', edakje, 3)
print('')

uitvoer = 'Waarden voor vereffende waarneming ydakje = A * xdakje + a0:'
print(uitvoer)
pm.print_m('ydakje', ydakje, 2)
print('')

# Controle
nullen = A.T * edakje

uitvoer= 'Controle op A.T * edakje = 0'
print(uitvoer)
pm.print_m('nullen', nullen, 3)
print('')

File: analyse/octave/opgave_16.m
# Opgave 16

addpath ("./lib")

# Gegeven
y = [78.83; 120.81; 38.71; 36.47; 80.61; 36.51] ;
A = [ 0 1 0 0;
      0 0 1 0;
      -1 1 0 0;
      0 0 -1 1;
      -1 0 1 0;
      0 0 -1 1;
] ;

# Vereffening volgens A-model
[ xdakje , ydakje , edakje ] = Amodel(A, y) ;

uitvoer=['Geschatte waarden voor parameters xdakje:'] ;
disp(uitvoer)
print_vector(xdakje, 'xdakje', 2) ;
disp('')

uitvoer=['Geschatte waarden voor toevallige afwijkingen edakje:'] ;
disp(uitvoer)
print_vector(edakje, 'edakje', 3) ;
disp('')

uitvoer=['Waarden voor vereffende waarneming ydakje = A * xdakje + a0:'] ;
disp(uitvoer)
print_vector(ydakje, 'ydakje', 2) ;
disp('')

# Controle
nullen = A' * edakje ;

uitvoer=['Controle op A'' * edakje = 0'] ;
disp(uitvoer)
print_vector(nullen, 'nullen', 3) ;
disp('')

